cads_adaptors/adaptors/mars.py

```diff
@@ -202,7 +202,11 @@ def retrieve_list_of_results(self, request: dict[str, Any]) -> list[str]:
         # Call normalise_request to set self.mapped_requests
         request = self.normalise_request(request)
 
-        data_formats = [req.pop("data_format", None) for req in self.mapped_requests]
+        # Invoke handle_data_format again as intersect_constraints may turn "data_format" into a list
```

Comment:
This should have been done in the previous data_formats PR, but I only found this instance when working on the intersect_constraints with MultiAdaptor entries.

Note this doesn't effect MultiMarsAdaptor entries, as they call execute_mars directly